tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodMetadataProvider.cs

```diff
@@ -69,14 +70,21 @@ public static void Remove(int index)
             }
         }
 
+        public static bool TryCreateIfNotExists(int index, in RuntimeMethodHandle methodHandle, in RuntimeTypeHandle typeHandle)
+        {
+            return TryCreateIfNotExists<object>(null, index, in methodHandle, in typeHandle);
+        }
+
         /// <summary>
         /// Tries to create a new <see cref="MethodMetadataInfo"/> at <paramref name="index"/>.
         /// </summary>
+        /// <param name="targetObject">The target object</param>
         /// <param name="index">The index of the method inside <see cref="_items"/></param>
         /// <param name="methodHandle">The handle of the executing method</param>
         /// <param name="typeHandle">The handle of the type</param>
+        /// <param name="asyncKickOffInfo">The info for the async kickoff method</param>
         /// <returns>true if succeeded (either existed before or just created), false if fails to create</returns>
-        public static bool TryCreateIfNotExists(int index, in RuntimeMethodHandle methodHandle, in RuntimeTypeHandle typeHandle)
+        public static bool TryCreateIfNotExists<TTarget>(TTarget targetObject, int index, in RuntimeMethodHandle methodHandle, in RuntimeTypeHandle typeHandle, AsyncHelper.AsyncKickoffMethodInfo asyncKickOffInfo = default)
```

Comment:
I'm a bit confused - is the `targetObject` different in each invocation of the method? If it is, we shouldn't store it in the `MethodMetadataInfo`, as it is not meant to maintain any mutable state.

Reply:
We don't store it in the `MetadataInfo`, only use it in `GetHoistedLocalsFromStateMachine`. Do you think we should do it in another way?

Reply:
Ah! That's fine then, thanks for the clarification. Resolving.